### PR TITLE
Add New Automation Workflows

### PR DESCRIPTION
**Pull request by nasbench**
### Summary of the Pull Request

This PR adds some automation workflows to make it easier for maintainers to maintain the rules and other quality of life features

### Changelog

chore: update workflows and add quality of life updates and automation to the repository

### Details of the Pull Request

- Enforces the following Yaml linter options as suggested in #4260 and  #4261
  - No trailing spaces
  - Required space after comment
  - Required new line at the EOF
  - Indentation is enforced to 4
  
- Add bot that automatically adds labels to PRs based on files being modified

- Add a greeter bot that will write a welcome message for new contributors for PRs and reminds them to read the conventions and stuff

- Implements the script proposed in #4473 and integrates it as a workflow that will be run twice a month and create a GH issue with the results.

- Implements a new version of Frack rule promotion script that upgrade rules status from `experimental` to `test` for rules that haven't been modified in 300+ days. The script run monthly in a workflow and creates a PR if there are modified files.

- Update the sigma tests workflow to be run only if rules are modified

- Added manual run options to most of the workflows

- Added `CODE_OF_CODUCT.md` and `CONTRIBUTING.md` files in order to ease the process of contributions and setup for the README update in the coming days

### Example Log Event

N/A

### Fixed Issues

N/A

### SigmaHQ Rule Creation Conventions

- If your PR adds new rules, please consider following and applying these [conventions](https://github.com/SigmaHQ/sigma-specification/blob/main/sigmahq/sigmahq_conventions.md)

### CODE REVIEW

**Review by frack113 — APPROVED** [review]
LGTM

File: tests/promote_rules_status.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+                ).days
+                if difference >= 300:
+                    rules_to_promote.append(file)
+    return rules_to_promote
+
+
+def promote_rules(rules_to_promote):
+    for file_ in rules_to_promote:
+        with open(file_, "r", encoding="utf8") as f:
+            data = f.read().replace("status: experimental", "status: test")
```
Comment (phantinuss) [inline tests/promote_rules_status.py:78]: Can we bind this regex style to "beginning of line / `^`"? Just as a precaution?

```
import "re"
[...]
re.sub("^status: experimental", "status: test", f.read())
```

(not sure it works exactly like that. Maybe f.read() needs to be placed in a temporary variable)

Comment (nasbench) [reply tests/promote_rules_status.py:78]: Read returns the content of the file as a string. I can prefix it with `\n` to make sure its from the start of the line.

If we want to use regex as you suggested I need to read the file as YAML and grab it as a dict or other similar thing.

The downside of this is the indentation can be lost during this conversion hence why I used `read` in the first place.

Let me know if the prefix idea is good enough.

Comment (phantinuss) [reply tests/promote_rules_status.py:78]: yes, using  `\n` as an anchor in that case does the same as `^` would.

**Review by phantinuss — APPROVED** [review]

**Pull request merged**